RELEASE/scripts/autoscend.ash

```diff
@@ -4449,77 +4449,97 @@ boolean LX_bitchinMeatcar()
 		{
-			meatRequired += npc_price(it);
+			cli_execute("make bitch");
+			cli_execute("place.php?whichplace=desertbeach&action=db_nukehouse");
+			return true;
+		}
+		else
+		{
+			auto_log_info("I got all the parts but I do not have enough meat to build a meatcar... doing something else", "red");
+			return false;
 		}
 	}
 
+	//if in knoll sign and thus has access to knoll store, then buy the meatcar components and then craft the meatcar if you can afford it
 	if(knoll_available())
 	{
-		if(my_meat() >= meatRequired)
+		int meatRequired = 0;
+		if(item_amount($item[Meat Stack]) == 0)
 		{
-			cli_execute("make bitch");
-			cli_execute("place.php?whichplace=desertbeach&action=db_nukehouse");
-			return true;
+			meatRequired += 100;
 		}
-		return false;
-	}
-	else
-	{
-		if((my_meat() >= (npc_price($item[Desert Bus Pass]) + 1000)) && isGeneralStoreAvailable())
+		foreach it in $items[Spring, Sprocket, Cog, Empty Meat Tank, Tires, Sweet Rims]
 		{
-			auto_log_info("We're rich, let's take the bus instead of building a car.", "blue");
-			buyUpTo(1, $item[Desert Bus Pass]);
-			if(item_amount($item[Desert Bus Pass]) > 0)
+			if(item_amount(it) == 0)
 			{
-				return true;
+				meatRequired += npc_price(it);
 			}
 		}
-		if(in_zelda()) return false;
-		auto_log_info("Farming for a Bitchin' Meatcar", "blue");
-		if(get_property("questM01Untinker") == "unstarted")
-		{
-			visit_url("place.php?whichplace=forestvillage&preaction=screwquest&action=fv_untinker_quest");
-		}
-		if((item_amount($item[Tires]) == 0) || (item_amount($item[empty meat tank]) == 0) || (item_amount($item[spring]) == 0) ||(item_amount($item[sprocket]) == 0) ||(item_amount($item[cog]) == 0))
+		if(my_meat() >= meatRequired)
 		{
-			if(!autoAdv(1, $location[The Degrassi Knoll Garage]))
-			{
-				if(guild_store_available())
-				{
-					visit_url("guild.php?place=paco");
-				}
-				else
-				{
-					abort("Need to farm a Bitchin' Meatcar but guild not available.");
-				}
-			}
-			if(item_amount($item[Gnollish Toolbox]) > 0)
-			{
-				use(1, $item[Gnollish Toolbox]);
-			}
+			cli_execute("make bitch");
+			cli_execute("place.php?whichplace=desertbeach&action=db_nukehouse");
+			return true;
 		}
 		else
 		{
-			if(my_meat() >= meatRequired)
-			{
-				cli_execute("make bitch");
-				cli_execute("place.php?whichplace=desertbeach&action=db_nukehouse");
-				return true;
-			}
+			auto_log_info("I do not have enough meat to build a meatcar... doing something else", "red");
 			return false;
 		}
 	}
-	return true;
+	
+	//if rich then just buy the desert pass
+	if((my_meat() >= (npc_price($item[Desert Bus Pass]) + 1000)) && isGeneralStoreAvailable())
+	{
+		auto_log_info("We're rich, let's take the bus instead of building a car.", "blue");
+		buyUpTo(1, $item[Desert Bus Pass]);
+		if(item_amount($item[Desert Bus Pass]) > 0)
+		{
+			return true;
+		}
+	}
+	
+	//plumbers should just wait until they are rich enough to buy the desert pass
```

Comment:
Stupid question incoming: why do this in Plumber but not Vampyre, or other avatar paths?

Reply:
This except was already there when I started reshuffling the meatcar logic.
I just reshuffled it to a more logical location and added the comment explaining what it is doing.

My guess is that this is because plumber
1. uses its own special currency (coins) for most things, so it has excess meat
2. has poor +item drops skills

so it somewhat makes sense to refuse to spend adventures on meatcar and save up for a bus pass. Although I am a bit leery of this optimization code, I decided to not delete it without first discussing it with people.

Reply:
Okay, so according to git blame I wrote that code but I'm genuinely not sure why. I suspect it was a hacky work-around to the moonsign issues you're fixing here, because I remember picking really weird moonsigns for my first few plumber runs. You don't need to change it in this PR, but I think you've uncovered a bug/suboptimality that I added that we can remove later. Whoopsie. Thanks!

Reply:
That makes sense. It probably really was to work around the issue I was trying to fix with weird path + sign combos failing the overly convoluted logic that was there before.

Do you want me to just remove this plumber exception as part of this pull? I don't mind doing so now.